### ISSUE TITLE
[Cop request]: UnneededRequireFileExtension

### ISSUE THREAD

**Issue by TSMMark** [rubocop-hq/rubocop#7333]
## Is your feature request related to a problem? Please describe.

`.rb` is not necessary when `require` or `require_relative` a Ruby file. When performing code review I am having to ask contributors to remove the superfluous file extension.

## Describe the solution you'd like

When given:

```ruby
require_relative "./whatever.rb"
require "../foo/bar.rb"
```

Autocorrect to:

```ruby
require_relative "./whatever"
require "../foo/bar"
```

**Comment by TSMMark** [comment]
I would be happy to implement this cop myself given a pointer or two.

**Comment by koic** [comment]
It seems to be good overall to me.

It is a caution.  `*.rb`, `*.o`, `*.so`, and other extensions are searched if the extension is omitted. (`*.rb` takes precedence) .
https://ruby-doc.org/core-2.6.4/Kernel.html#method-i-require

There is an edge case where `foo.so` file is loaded instead of a `LoadError` if `foo.so` file exists when `require 'foo.rb'` will be changed to `require 'foo'`.

FYI, the cop name should use `Redundant` instead of `Unneeded`. And I think it's probably the Style department, not the Lint department.
https://github.com/rubocop-hq/rubocop/issues/7077

@rubocop-hq/rubocop-core Is there anything you are worrying about?

**Comment by jonas054** [comment]
It looks safe to me. Since `*.rb` take precedence I can't see any risk that removing `.rb` would alter the behavior, except for the edge case @koic mentioned, but that seems harmless.

**Comment by jonas054** [comment]
@TSMMark Start by reading [Contributing](https://github.com/rubocop-hq/rubocop/blob/master/CONTRIBUTING.md) and let us know if you have any questions. For this cop I think you should implement `on_send` and check calls to `require` and `require_relative`. Call the cop `Style/RedundantFileExtensionInRequire`.

**Comment by TSMMark** [comment]
Sorry y'all I totally lost track of this ticket! @fatkodima you rock!
